crates/env/src/engine/on_chain/impls.rs

```diff
@@ -726,4 +727,40 @@ impl TypedEnvBackend for EnvInstance {
         let enc_code_hash = scope.take_encoded(code_hash);
         ext::unlock_delegate_dependency(enc_code_hash)
     }
+
+    fn xcm_execute<E, Call>(&mut self, msg: &VersionedXcm<Call>) -> Result<()>
+    where
+        E: Environment,
+        Call: scale::Encode,
+    {
+        let mut scope = self.scoped_buffer();
+
+        // Double encoding the message as the host fn expects an encoded message.
+        let enc_msg = scope.take_encoded(&scale::Encode::encode(msg));
+        #[allow(deprecated)]
```

Comment:
unstable host api (which is still the case for this one as of this release of polkadot-sdk) are marked as deprecated so that they can trigger a warning if you try to use them without specifying `allow(deprecated)`